### PR TITLE
throw exception when slicing a 0-dim tensor

### PR DESCRIPTION
**Pull request by yxlao**
This change is [Reviewable](https://reviewable.io/reviews/isl-org/open3d/4043)

### CODE REVIEW

File: python/test/core/test_core.py [hunk 1]
```diff
@@ -440,6 +440,12 @@ def test_getitem(device):
     np.testing.assert_equal(o3_t[0:2, 1:3, 0:4][0:1, 0:2, 2:3].cpu().numpy(),
                             np_t[0:2, 1:3, 0:4][0:1, 0:2, 2:3])
 
+    # Slice a zero-dim tensor
+    with pytest.raises(Exception):
```
Comment (ssheorey) [inline python/test/core/test_core.py:444]: `with pytest.raises(RuntimeError, match="Cannot slice a scalar (0-dim) tensor.")`
(more specific exception class with error message)
also below...

File: python/test/core/test_core.py [hunk 2]
```diff
@@ -548,6 +554,12 @@ def test_setitem(device):
     o3_t[2, 2] = False
     np.testing.assert_equal(o3_t.cpu().numpy(), np_t)
 
+    # Slice a zero-dim tensor
+    with pytest.raises(Exception):
+        o3c.Tensor.ones((), device=device)[:] = 0
+    with pytest.raises(Exception):
+        o3c.Tensor.ones((), device=device)[0:1] = 0
+
```
Comment (ssheorey) [inline python/test/core/test_core.py:562]: More specific test here as well.

**Review by ssheorey — APPROVED** [review]

File: python/test/core/test_core.py (hunk 1 above)
Comment (yxlao) [reply python/test/core/test_core.py]: done

File: python/test/core/test_core.py (hunk 2 above)
Comment (yxlao) [reply python/test/core/test_core.py:567]: done

**Pull request merged**